### PR TITLE
Git cli home env

### PR DESCRIPTION
**Pull request by khensel17**
# Changes

Fix nonroot user support: setting HOME env

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: task/git-cli/0.4/git-cli.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      type: string
+      default: |
+        git help
+
+    - name: USER_HOME
+      description: |
+        Absolute path to the user's home directory. Set this explicitly if you are running the image as a non-root user or have overridden
+        the gitInitImage param with an image containing custom user configuration.
+      type: string
+      default: "/root"
```
Comment (vinamra28) [inline task/git-cli/0.4/git-cli.yaml:82]: can you please make an entry of the same in README as well?

**Review by vinamra28 — COMMENTED** [review]
also can you please squash the commits and then we can merge this 🙇

**Comment by khensel17** [comment]
> also can you please squash the commits and then we can merge this 🙇

Commits are now squashed 😄

**Comment by vinamra28** [comment]
/lgtm
/approve
Thanks @khensel17 🙂

**Comment by khensel17** [comment]
@vinamra28 I fixed some linting errors. Can you please provide the approve and lgtm labels again? :)

**Comment by vinamra28** [comment]
> @vinamra28 I fixed some linting errors. Can you please provide the approve and lgtm labels again? :)

@khensel17, I can see that you have made some changes in the script, can you please update the tests to include a check whether the results are being produced accurately or not? (Just to make sure that this doesn't breaks)

**Comment by khensel17** [comment]
@vinamra28 The test is updated to check the result of the git-cli task.

File: task/git-cli/0.4/tests/run.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      params:
+      - name: expected-commit
+      workspaces:
+      - name: source
+      steps:
+      - image: alpine/git:v2.36.2
+        script: |
+          #!/usr/bin/env sh
+          cd $(workspaces.source.path)
+          receivedCommit=$(git rev-parse HEAD)
```
Comment (vinamra28) [inline task/git-cli/0.4/tests/run.yaml:58]: probably this can be taken from the result `commit` produced by the `git-cli` Task. But this is also fine 😅

**Review by vinamra28 — APPROVED** [review]
/lgtm

**Pull request merged**